api/src/main/java/org/apache/iceberg/TableScan.java

```diff
@@ -101,4 +101,11 @@ default TableScan appendsAfter(long fromSnapshotId) {
    * @return the Snapshot this scan will use
    */
   Snapshot snapshot();
+
+  /**
+   * Create a new {@link TableScan} from this scan's configuration that will have column stats
+   *
+   * @return a new scan based on this with column stats
+   */
+  TableScan withColStats();
```

Comment:
Rather than adding a revapi change, can you add a default implementation that throws `UnsupportedOperationException`? We want to avoid a breaking API change.

Reply:
Done